app/services/courses/generate_course_title_service.rb

```diff
@@ -0,0 +1,54 @@
+module Courses
+  class GenerateCourseTitleService
+    def execute(course:)
+      subjects = course.subjects
+
+      title = if course.further_education_course?
+                further_education_title
+              elsif is_modern_lanuage_course?(subjects)
+                modern_language_title(subjects)
+              else
+                generated_title(subjects)
+              end
+
+      title = append_send_info(title) if course.is_send?
+      title
+    end
+
+  private
+
+    def is_modern_lanuage_course?(subjects)
+      subjects[0] == SecondarySubject.modern_languages
+    end
+
+    def modern_language_title(subjects)
+      title = "Modern Languages"
+
+      if subjects.length == 2
+        title += " (#{subjects[1]})"
+      elsif subjects.length == 3
+        title += " (#{subjects[1]} and #{subjects[2]})"
+      end
+
+      title
+    end
+
+    def generated_title(subjects)
+      return "" if subjects.empty?
+
+      if subjects.length == 1
+        subjects[0].to_s
+      else
+        "#{subjects[0]} with #{subjects[1]}"
```

Comment:
We **MUST** add these tests cases

https://github.com/DFE-Digital/search-and-compare-ui/blob/903e35ea800ccc4cf0b06a18a36e037ab813e229/tests/Unit.Tests/ViewFormatters/CourseExtensions.Tests.cs#L145

its related to 
https://github.com/DFE-Digital/search-and-compare-ui/blob/903e35ea800ccc4cf0b06a18a36e037ab813e229/shared/ViewFormatters/CourseExtensions.cs#L149

if title starts with any of the subject
```
                "Drama",
                "Media Studies",
                "PE",
                "Physical Education",
```

then there is no financial incentives

fyi @davidgisbey what was discussed earlier.

Reply:
Hm - So my concern with this is similar to a comment you had elsewhere

Do we have any consistency in what we deem a courses "Primary" subject and it's "Secondary" subject. I know in Find we had to base it off the title, but I'm concerned the ordering could be inconsistent during creation in the array and it causes issues with bursaries further down the line :thinking:

Reply:
Find don't care about that as it was never properly define. By ordinal or weight, ot string parsing.